### DISCUSSION TITLE
GeoIP DB / Clouflfare real IP in caprover service / nginx

### DISCUSSION THREAD

**Discussion by fsennj** [caprover/caprover#1309] [Q&A]
Hello all, is it possbile to use the GEO IP for certain services in caprover?

https://fedingo.com/how-to-block-ip-by-country-in-nginx/

and also any idea for SSL Certifactes with Cloudflare DNS/Caching to retrieve the real IP so renewing Certs gets easier?

**Comment by githubsaturn** [comment]
> Hello all, is it possbile to use the GEO IP for certain services in caprover?

If you want additional packages to be installed in your nginx container, you can build a custom nginx container, and use [config override](https://caprover.com/docs/troubleshooting.html#customize-config-settings) to override the nginxImageName value.

https://github.com/caprover/caprover/blob/6fdcb8b1e4aa4e2acafb02ec5914b3e9bfba4c4e/src/utils/CaptainConstants.ts#L43-L45


Regarding real IP, it really depends on the upstream services that you're using, but conventionally they all must have `x-forwarded-for` header that contains the original IP.
